### ISSUE TITLE
Cannot read property '0' of undefined error when I tried jhipster-uml with my xmi

### ISSUE THREAD

**Issue by abrahamjins** [jhipster/jhipster-uml#232]
### The issue
I am getting the below error when I tried jhipster-uml.
Parser detected: MODELIO.

An error has occurred:
        TypeError
Error message:
        Cannot read property '0' of undefined
Stack trace:
TypeError: Cannot read property '0' of undefined
    at addAssociation (C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\editors\modelio_parser.js:156:38)
    at fillAssociations (C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\editors\modelio_parser.js:150:5)
    at Object.parse (C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\editors\modelio_parser.js:28:3)
    at Object.<anonymous> (C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\jhipsteruml.js:54:25)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)

Kindly help me to sort this out.My xmi is bit large. Please find the attached xmi.

---
### The command

| jhipster-uml trackz_new.xmi |

---
#### The stacktrace
Parser detected: MODELIO.

An error has occurred:
        TypeError
Error message:
        Cannot read property '0' of undefined
Stack trace:
TypeError: Cannot read property '0' of undefined
    at addAssociation (C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\editors\modelio_parser.js:156:38)
    at fillAssociations (C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\editors\modelio_parser.js:150:5)
    at Object.parse (C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\editors\modelio_parser.js:28:3)
    at Object.<anonymous> (C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\jhipsteruml.js:54:25)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)

### Steps to reproduce

---

### The XMI file
 I have emailed my xmi to below email id.

---

### Versions 

| JHipster           | JHipster UML           |
| ------------------ | ---------------------- |
| `JHipster Version` |    `jhipster-uml -v`   |
 The current version of JHipster UML is 2.0.0

**Comment by MathieuAA** [comment]
Hi, I'll look into it later today.

Le 8 novembre 2016 06:44:36 GMT+01:00, abrahamjins <notifications@github.com> a écrit :
>
>
>### The issue
>I am getting the below error when I tried jhipster-uml.
>Parser detected: MODELIO.
>
>An error has occurred:
>        TypeError
>Error message:
>        Cannot read property '0' of undefined
>Stack trace:
>TypeError: Cannot read property '0' of undefined
>at addAssociation
>(C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\editors\modelio_parser.js:156:38)
>at fillAssociations
>(C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\editors\modelio_parser.js:150:5)
>at Object.parse
>(C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\editors\modelio_parser.js:28:3)
>at Object.<anonymous>
>(C:\Users\pc\AppData\Roaming\npm\node_modules\jhipster-uml\lib\jhipsteruml.js:54:25)
>    at Module._compile (module.js:413:34)
>    at Object.Module._extensions..js (module.js:422:10)
>    at Module.load (module.js:357:32)
>    at Function.Module._load (module.js:314:12)
>    at Module.require (module.js:367:17)
>    at require (internal/module.js:16:19)
>
>Kindly help me to sort this out.My xmi is bit large. Please find the
>attached xmi.
>
>
>---
>### The command
>
>
>| `Replace the content` |
>---
>#### The stacktrace
>
>
>```
>Paste it here!
>```
>
>---
>
>### Steps to reproduce
>
>
>
>---
>
>### The XMI file
> I have emailed my xmi to below email id.
>
>
>---
>
>### Versions 
>
>| JHipster           | JHipster UML           |
>| ------------------ | ---------------------- |
>| `JHipster Version` |    `jhipster-uml -v`   |
>
>
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/jhipster/jhipster-uml/issues/232

-- 
Mathieu ABOU-AICHI

**Comment by MathieuAA** [comment]
I can't even import your XMI in modelio... What version of Modelio were you using for this model? 
I've also found something really weird: you have a field that has the same ID as an association.

**Comment by MathieuAA** [comment]
Moreover, your XMI model seems to have integrity issues...

**Comment by abrahamjins** [comment]
Modelio Version 3.5.01

**Comment by MathieuAA** [comment]
I've tried again, but I can't import it at all. There are too many errors preventing it from being imported.

**Comment by abrahamjins** [comment]
I have sent you an updated xmi. I was able to import it correctly.

Thank you.

**Comment by MathieuAA** [comment]
Thanks, I'll test it out as soon as I receive it.

**Comment by MathieuAA** [comment]
Your XMI is still invalid... I don't know if it's a bug from v3.5.01 of Modelio, but there's still the ID problem.
Try with the latest version of Modelio, because I can't debug it.

**Comment by abrahamjins** [comment]
Thanks Mathieu.Am getting error when I imported the xmi in modelio 3.6.
When I tried to open my project(created in 3.5) in modelio 3.6, I am getting below error.
The 'PredefinedTypes 3.5.03' fragment's 'Standard v1.1.00' metamodel is outdated compared to the current 'Standard v2.0.00' version
Am not getting any clues on this. kindly let me know your thoughts.

**Comment by MathieuAA** [comment]
I've actually succeeded in finding and downloading the exact version you were using, and the import was successful. I'll try to see where it failed and what the problem is... About your type problem, they may have made some breaking changes... which is absolutely stupid.

**Comment by MathieuAA** [comment]
The migration works for me, but I'm not sure about what has changed between v3.5.X and v3.6. Moreover, there seems to be some heavy breaking changes between v3.6 and everything before.
I've got a question, did you only use binary relationships? Or does your diagram have ternary ones?
Finally, I think the issue consists in invalid relationships:
```
<packagedElement xmi:type="uml:Association" xmi:id="_DcWlz6gIEeaRRbO6lIVTnQ" name="CampFacility" memberEnd="_DcWlxqgIEeaRRbO6lIVTnQ _DcWl1KgIEeaRRbO6lIVTnQ"/>
<packagedElement xmi:type="uml:Association" xmi:id="_DcWl0KgIEeaRRbO6lIVTnQ" name="RoomFacility" memberEnd="_DcWlyagIEeaRRbO6lIVTnQ _DcWl_KgIEeaRRbO6lIVTnQ"/>
```
Modelio V3.5.X may have introduced this strange and anormal behavior, and fixed it in v3.6 because I couldn't reproduce the same thing in v3.3, v3.4 and v3.6. For you information, it only occurs with bidirectional Many-to-Many relationships.
